src/pages/article/[slug].tsx

```diff
@@ -0,0 +1,159 @@
+    row-gap: ${theme.spacings.xxsmall};
+  `}
+`
+
+const Title = styled.h1`
+  ${({ theme }) => css`
+    font-size: ${theme.fonts.sizes.xhuge};
+    color: ${theme.colors.black[100]};
+    line-height: 47px;
+    letter-spacing: -0.04em;
+  `}
+`
+
+const Description = styled.h3`
+  ${({ theme }) => css`
+    font-size: ${theme.fonts.sizes.xmedium};
+    color: ${theme.colors.grey[200]};
+    line-height: ${theme.spacings.xxmedium};
+    font-weight: 400;
+  `}
+`
+
+const Article: NextPage<ArticleNextPageProps> = ({ slug }) => {
+  const { data, refetch, isFetching } = useGetArticle(slug)
+
+  const dataOption = fromNullable(data)
+
+  return f(() => {
+    if (isSome(dataOption)) {
+      if (isRight(dataOption.value)) {
+        return (
+          <>
+            <ArticleSeo
+              author={dataOption.value.right.article.author.username}
+              description={dataOption.value.right.article.description}
+              modifiedTime={dataOption.value.right.article.updatedAt}
+              publishedTime={dataOption.value.right.article.createdAt}
+              slug={dataOption.value.right.article.slug}
+              title={dataOption.value.right.article.title}
+              tags={dataOption.value.right.article.tagList}
+            />
+            <Wrapper>
+              <HeaderSection>
+                <ArticleHeader
+                  date={dataOption.value.right.article.createdAt}
+                  name={dataOption.value.right.article.author.username}
+                  readTime={dataOption.value.right.article.updatedAt}
+                />
+                <Title>{dataOption.value.right.article.title}</Title>
+                <Description>
+                  {dataOption.value.right.article.description}
+                </Description>
+              </HeaderSection>
+              <ArticleBody articleText={dataOption.value.right.article.body} />
+            </Wrapper>
+          </>
+        )
+      } else {
+        return (
+          <ErrorState
+            message="Something went wrong while trying to requesting the user informations."
+            title="Something went wrong"
+            buttonLabel="Try again"
+            onButtonClick={refetch}
+            isButtonLoading={isFetching}
+            disabled={isFetching}
+          />
+        )
+      }
+    } else {
+      return (
+        <ErrorState
+          message="Something went wrong while trying to requesting the user informations."
+          title="Something went wrong"
+          buttonLabel="Try again"
+          onButtonClick={refetch}
+          isButtonLoading={isFetching}
+          disabled={isFetching}
+        />
+      )
+    }
+  })
+}
+
+interface GetServerSidePropsParams extends ParsedUrlQuery {
+  slug: string
+}
+
+export const getServerSideProps = async ({
+  params,
+}: GetServerSidePropsContext<GetServerSidePropsParams>) => {
+  const queryClient = new QueryClient()
+
+  const paramsOption = fromNullable(params)
```

Comment:
Porque não usar pipe aqui e na função de baixo?

Reply:
What do you think about this [way](https://github.com/joaooliveiradev/conduit-frontend/pull/40/commits/efd2f94c4f8d99167e259430eb2bc7a0d58cfaf2)